### ISSUE TITLE
Documenting acceptance criteria

### ISSUE THREAD

**Issue by dferber90** [commercetools/ui-kit#248]
TL;DR: We want to add `ACCEPTANCE_CRITERIA.md` files next to `README.md` files which contain the acceptance criteria. They should be visible through storybook. They should be the single source of truth for how the component should behave. They aim to make it easier to learn about the intended behaviour of components.

## Problem

When making changes to an existing component it is inconvenient/hard/a lot of work to collect the acceptance criteria it was built on. This makes changing the component hard as the original requirements are hard to find. Modifications can therefore lead to losing functionality which was part of previous acceptance criteria.

## Current situation

Right now our acceptance criteria is located in Jira. When a new story is started (to build a new component), the acceptance criteria is collected there. After building the component, the acceptance criteria is not anywhere and just resides there. Sometimes when new things become evident along the way, additional information is mentioned in comments but doesn't become part of the acceptance criteria as it is not updated.

When bugs are found, new stories are created. Sometimes fixing bugs requires making decisions which change the original acceptance criteria. The original story is never updated and the additional acceptance criteria gets added to the new bug story.

To find the currently applied acceptance criteria, somebody needs to dig through all stories and bug reports of that component by searching for its name in Jira (and by following any "relates-to" links and by following subtasks of the stories). When a component was renamed in the process (e.g. while moving it from `core` to ui-kit) then Jira has to be searched for the old name of that component as well.

This makes it really hard to collect the current acceptance criteria which the component is built on, as
- the acceptance criteria is not colocated with the code (it is in Jira instead)
- the acceptance criteria is spread through multiple stories
- the acceptance criteria is not always updated after verbally agreeing on something

This can get really cumbersome and I therefore haven't seen us taking previous acceptance criteria into consideration when working on components. Nor do we write in-depth acceptance criteria to begin with.

### Colocating

We want to have one acceptance criteria document per component which serves as the single source of truth. For every component, aside from having a `README.md` we also want to have an `ACCEPTANCE_CRITERIA.md` next to it. This makes it easy to access while working on the code.

### Making acceptance criteria visible

Having a markdown document tos how the acceptance criteria is nice for developers, but we also need to show the documentation to designers and QA. Our suggestion is to make the documentation accessible by displaying the `ACCEPTANCE_CRITERIA.md` from the codebase in Storybook by adding another tab next to the readme, like this:

![bildschirmfoto 2018-11-23 um 12 08 36](https://user-images.githubusercontent.com/1765075/48940633-afa6d500-ef18-11e8-9236-f81ba053e535.png)

### Adding acceptance criteria when adding new components

The acceptance criteria would still start out in a Jira story when building a new component while the component is being refined. The PR which adds the component usually contains the documentation for it (the `README.md` file). Going forward, we should also add an `ACCEPTANCE_CRITERIA.md` file to the PR which contains the final acceptance criteria the component was built with.

### Changing acceptance criteria when changing components

In case a bug is found, or the behaviour of a component changes (and the acceptance criteria with it), the new behaviour can be discussed in Jira or verbally. The PR which changes the code then needs to change the `ACCEPTANCE_CRITERIA.md` file as well. This aims to keep the `ACCEPTANCE_CRITERIA.md` and the actual behaviour of the component in sync. 

All this aims to make it easier to learn about the intended behaviour of built components.

---

We came up with this suggestion after running into this problem. Thanks to @mariabarrena @filippobocik  @antonboyko and @coolmilo who helped form this approach 💐
